### DISCUSSION TITLE
(Capstone DV) Korelasi dari data frame yang berbeda-beda?

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#565] [Q&A] [answered]
Saya ingin melihat plot korelasi untuk banyak saham secara sekaligus, namun data setiap saham terpisah dalam data frame yang berbeda-beda. Setiap data frame saham memiliki kolom sebagai berikut:
![image](https://user-images.githubusercontent.com/122977451/220619788-3db69c3e-f034-4644-bd31-31bd43ccb882.png)

Saya ingin melihat korelasi antara kolom DR (Daily Return saham) dari masing-masing data frame saham, yang dimana pada hasil akhirnya user dapat melihat korelasi antara dua saham yang mereka pilih, namun kurang mengerti cara menggabungkan data-data tersebut kedalam scatter plot ggplot. Sebagai gambaran besar, X-axis dan Y-axis adalah berdasarkan DR dari saham yang dibandingkan, ilustrasi sebagai berikut:
![image](https://user-images.githubusercontent.com/122977451/220623633-00620747-709a-491a-941a-5a4ec61cbc25.png)

Sebagai informasi tambahan, saya juga sudah mencoba menyatukan semua data frame saham yang saya gunakan dan mencoba melakukan hal yang sudah saya tuliskan diatas, namun hasilnya malah seperti dibawah ini:
![image](https://user-images.githubusercontent.com/122977451/220621906-9847ea3b-7b27-4690-a2ef-55a0dfe38710.png)

Terima kasih sebelumnya.

**Comment by finesaaa** [comment]
Dear @AGelfP,
Terima kasih atas pertanyaannya, mohon ditunggu untuk balasan dari tim kami, akan diproses

**Comment by idoali** [comment, marked as answer]
Terima kasih atas pertanyaannya pak

Data yang ingin ditampilkan dengan plot harus ada di dataframe yang sama pak. Tidak boleh terpisah. 

Saran yang bisa kami berikan ke Bapak sekarang adalah untuk membuat satu dataframe yang mengandung semua harga saham yang ingin bapak tampilkan. Jadi pada dataframe ini kita bisa menemukan harga saham ADMF dan saham-saham lainnya. 

Ada banyak fungsi di R yang bisa bapak gunakan untuk melakukan ini. Salah satunya adalah `left_join` dari `dplyr` . Dengan fungsi ini, kita bisa menggabungkan dua dataframe dengan menjadikan salah satu kolom menjadi acuannya. 

Jika sedikit membingungkan, dengan membuat code nya langsung kemungkinan bapak akan mengerti. Katakanlah bapak punya dua dataframe, `admf` dan `idx`. Dua dataframe ini punya nama kolom yang sama. Kita akan membuat dataframe baru `df` yang berisikan kolom `Date` `DR.ADMF` dan `DR.IDX` dengan `left_join`. Berikut caranya :

```{r}
df <- left_join(admf[, c("Date", "DR")], idx[, c("Date", "DR")], 
                by = Date)
```
Untuk penjelasan lebih lengkapnya terkain `left_join` bisa dilihat di dokumentasinya pak [https://dplyr.tidyverse.org/reference/mutate-joins.html](https://dplyr.tidyverse.org/reference/mutate-joins.html)

Hasilnya nanti bapak akan mendapatkan dataframe dengan nama kolom `Date`, `DR.x`, dan `DR.y`. Bapak bisa mengubah nama dari`DR.x`, dan `DR.y` menjadi `admf` dan `idx` . Langkah berikutnya adalah bapak bisa menggunakan fungsi `pivot_longer` dimana bapak akan menampilkan semua harga baik dari admf maupun dari idx di kolom yang sama. Berikut kurang lebih caranya :

```{r}
df <- pivot_longer(df, cols = c(2, 3), names_to = "company", values_to = "dr")
```
Untuk contoh lengkapnya bisa dilihat di dokumentasi juga pak https://tidyr.tidyverse.org/reference/pivot_longer.html . Sesudah dari sini, bapak sudah punya satu dataframe yang siap untuk dibuat visualisasinya.

**Reply by AGelfP** [reply]
lagi saya coba Kak, muncul error seperti ini
![image](https://user-images.githubusercontent.com/122977451/220828712-2cfe1352-2747-4d51-8cc6-c7d5e16f8b94.png)

EDIT: Sudah aman Kak, ternyata di by = Date, harus ditulis by = "Date"

**Comment by idoali** [comment]
Date harus diubah jadi karakter pak. Jadi `"Date"` bukan `Date`
